src/nvim/misc1.c

```diff
@@ -2916,10 +2916,16 @@ char_u *vim_getenv(char_u *name, int *mustfree)
 {
   char_u      *p;
   char_u      *pend;
-  int vimruntime;
+  bool env_vimruntime = (STRCMP(name, "VIMRUNTIME") == 0 || STRCMP(name, "NVIMRUNTIME") == 0);
+  bool env_vim = (STRCMP(name, "VIM") == 0 || STRCMP(name, "NVIM") == 0);
 
-
-  p = (char_u *)os_getenv((char *)name);
+  if (env_vimruntime) {
+    p = (char_u *)os_getenv("NVIMRUNTIME");
+  } else if (env_vim) {
+    p = (char_u *)os_getenv("NVIM");
+  } else {
+    p = (char_u *)os_getenv((char *)name);
+  }
```

Comment:
In here, and in vim_getenv as well, whenever VIMRUNTIME is passed it is converted to NVIMRUNTIME i.e. VIMRUNTIME is never actually set and NVIMRUNTIME is set instead.

The same applies for VIM-NVIM.